rocketpool-daemon/common/validator/validator-manager.go

```diff
@@ -272,3 +284,139 @@ func (m *ValidatorManager) checkIfReady() error {
 	}
 	return utils.CheckIfWalletReady(status)
 }
+
+func (m *ValidatorManager) GetWalletStatus() (walletcore.WalletStatus, error) {
+	status, err := m.wallet.GetStatus()
+	if err != nil {
+		return status, err
+	}
+	if !walletcore.IsWalletReady(status) {
+		return status, fmt.Errorf("wallet is not ready")
+	}
+	return status, nil
+}
+
+func (m *ValidatorManager) InitializeBindings(status walletcore.WalletStatus) (*node.Node, *minipool.MinipoolManager, error) {
```

Comment:
this function returns a `node.Node` and a `minipool.MinipoolManager`, which are then passed in to `GetMinipools`.

This seems like a poor design choice: why not simply create those pointers in `NewValidatorManager` and store them as unexported fields on `ValidatorManager`? They can then be accessed in `GetMinipools` with `m.node` or `m.minipoolManager`